app/models/ability.rb

```diff
@@ -0,0 +1,36 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    #
+    user ||= User.new # guest user (not logged in)
+
+    if user.present?
+      can :manage, :all
+    elsif can :manage, Post, author_id: user.id
+      can :manage, Comment, author_id: user.id
+      can :read, :all
+    else
+      can :read, :all
+    end
+    #
```

Comment:
- [ ] Inside your User model, please create a method to check if a user is an admin or not:

```ruby
def admin?
     role == 'admin' ? true : false
end
```

Please, use this method every time you need to check if a user is an admin.

Please kindly remember:

- A user can delete a post if it is theirs or if they have an admin role
- A user can delete a comment if it is theirs or if they have an admin role